src/coffee/collector.coffee

```diff
@@ -0,0 +1,47 @@
+((window, $) ->
+  selectors = {
+    fileOpenListItem: 'a.file-list-link',
+    fileDownloadListItem: 'a.file-list-control',
+    videoPlayListItem: '.views-sidebar-video .playlist-list-item-link',
+    audioPlayListItem: '.views-sidebar-audio .playlist-list-item-link',
+    galleryListItem: '.gallery-list-item-link'
+  }
+
+  collectEvent = (path, type) ->
+    ($ window).trigger 'analytics-collect', [{path: path, type: type}]
+
+  fileOpenListItemClicked = (e) ->
+    el = $(@)
+    if el.data('mimetype') == 'html' 
+      collectEvent el.attr('href'), 'html'
+    else if el.data('type') == 'directory'
+      collectEvent el.attr('href'), 'folder'
+    else 
+      collectEvent el.attr('href'), 'file'
```

Comment:
I think, since we're already messing with mimetypes, it's simpler to use the first portion of the mime type (before the slash), instead of generic 'file'. Iirc, we had handling for video, audio, etc in the previous setup.